diffpy/snmf/tests/test_polynomials.py

```diff
@@ -11,14 +12,14 @@
     ([0, -7], 0),
     ([-9, 0], 3),
     ([-9, 3], 2.8169),
-    ([[2, 2], [2]], [0, 0]),
-    ([[[2, 2], [2, 2]], 2], [0, 0]),
-    ([[[[3,2],[-2,-2],[100,0]]],2],None)
+    ([[2, 2], 2], [0, 0]),
```

Comment:
you appear to have changed the tests and not the code to make the tests pass.  This is generally not a good idea because presumably your workflow is to make the tests to give you your desired behavior, then to write the code to make the tests pass. 

As an exercise, maybe put a comment after each test to elaborate what behavior that test is testing and we can discuss what behavior we actually want, code that into the test, then modify the code.

All your tests test lists of ints.  Is that intended or floats are also allowed?

Reply:
> Floats are also allowed. I changed the test cases due to an error that emerged when I updated python and the package's dependencies. One of the tests was incomplete, so it never passed it the first place. In line 14, the 2 was changed from a list to a scalar.

sure, this was a bad test, but the one that returns `None`, or doesn't is a "behavior".  Do you want your function to return `None` or `[0,0]` given those inputs, and why did you make that choice?